numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
I see. Yes, I would be that one who enable good optimization only on good platform. That's enough for the practical purpose. If compiler defines its version incorrectly - it is compiler bug, just do not support such a compiler. Otherwise, it is a bug in condition, e.g. when only compiler version is checked but the code really depends on libc version rather than compiler version.
I worked in a cross-platform project where all the configuration is done in the header config file. It is verbose, ugly, but portable and certainly feasible even in absence of features like `has_include` or `has_builtin`.

Anyway, thanks for the discussion, we'll open a separate issue on this.